File: 2594_minimum_time_to_repair_cars.py
import math

class Solution(object):
    def repairCars(self, r, c):
        l, h = 0, min(r) * c * c
        while l < h:
            m = (l + h) // 2
            if sum(int(math.sqrt(m // i)) for i in r) >= c:
                h = m
            else:
                l = m + 1
        return l

File: 2594_minimum_time_to_repair_cars.java
import java.util.*;

class Solution {
    public long repairCars(int[] r, int c) {
        long l = 0, h = (long) Arrays.stream(r).min().getAsInt() * c * c;
        while (l < h) {
            long m = (l + h) / 2, s = 0;
            for (int i : r) s += Math.sqrt(m / i);
            if (s >= c) h = m; else l = m + 1;
        }
        return l;
    }
}
